src/GridFS/ReadableStream.php

```diff
@@ -186,9 +186,26 @@ public function seek($offset)
         $this->chunkOffset = (integer) floor($offset / $this->chunkSize);
         $this->bufferOffset = $offset % $this->chunkSize;
 
-        if ($lastChunkOffset !== $this->chunkOffset) {
+        /* If we are seeking to a previous chunk, we need to reinitialize the
+         * chunk iterator.
+         */
+        if ($lastChunkOffset > $this->chunkOffset) {
             $this->buffer = null;
             $this->chunksIterator = null;
+            return;
+        }
+
+        /* If we are seeking to a subsequent chunk, we do not need to
+         * reinitalize the chunk iterator. Instead, we can simply move forward
+         * to $this->chunkOffset.
+         */
+        $numChunks = $this->chunkOffset - $lastChunkOffset;
+        $i = 0;
+        if ($this->chunksIterator !== null) {
+            while ($i < $numChunks) {
+                $this->chunksIterator->next();
```

Comment:
If we look at the previous code, both `$buffer` and `$chunksIterator` were reset when the chunk offset changed. When `readBytes()` is called, it initializes those properties again if they are `null`.

The current change in this PR leaves `$buffer` as-is so long as the current chunk offset is >= the previous offset. That becomes problematic if you do end up advancing to the next chunk, as you do here.

The test case in https://github.com/kvwalker/mongo-php-library/pull/5/files#diff-40e5981925700af63fd6c7a3417a12cdR294 demonstrates a potential failure where `$buffer` retains the contents of the first chunk and we end up reading "abcd" instead of "efgh".